pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/grpc/PipelinesTest.java

```diff
@@ -0,0 +1,585 @@
+package com.hedera.pbj.runtime.grpc;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Flow;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+class PipelinesTest {
+
+    @Nested
+    @ExtendWith(MockitoExtension.class)
+    class NoopTest {
+        @Mock Flow.Subscription subscription;
+
+        @Test
+        void noopWithNullSubscriptionThrowsNPE() {
+            final var noop = Pipelines.noop();
+            assertThat(noop).isNotNull();
+            assertThatThrownBy(() -> noop.onSubscribe(null))
+                    .isInstanceOf(NullPointerException.class);
+        }
+
+        @Test
+        void noopOnSubscribeCallsTheSubscriptionWithMaxRequests() {
+            final var noop = Pipelines.noop();
+            noop.onSubscribe(subscription);
+            verify(subscription).request(Long.MAX_VALUE);
+        }
+
+        @Test
+        void noopOnNextDoesNothing() {
+            final var noop = Pipelines.noop();
+            noop.onNext(Bytes.wrap("SHOULD NOT DO ANYTHING"));
+            assertThat(noop).isNotNull(); // if we get here, all is well.
+        }
+
+        @Test
+        void errorsCauseSubscriptionToBeCanceled() {
+            final var noop = Pipelines.noop();
+            noop.onSubscribe(subscription);
+            noop.onError(new RuntimeException("Causes the subscription to be cancelled"));
+            verify(subscription).cancel();
+        }
+
+        @Test
+        void noopOnCompleteDoesNothing() {
+            final var noop = Pipelines.noop();
+            noop.onComplete();
+            assertThat(noop).isNotNull(); // if we get here, all is well.
+        }
+
+    }
+
+    @Nested
+    @ExtendWith(MockitoExtension.class)
+    class UnaryTest {
+        @Mock Flow.Subscriber<Bytes> replies;
+        @Mock Flow.Subscription subscription;
+
+        @Test
+        void requestMapperIsRequired() {
+            final var builder = Pipelines.<String, String>unary()
+                    .method(String::toUpperCase)
+                    .mapResponse(Bytes::wrap)
+                    .respondTo(replies);
+
+            assertThatThrownBy(builder::build).isInstanceOf(IllegalStateException.class);
+        }
+
+        @Test
+        void methodIsRequired() {
```

Comment:
Might want to add `.hasMessage(...)` checks to these tests to differentiate between various IllegalStateExceptions.